src/models/ListRule.ts

```diff
@@ -15,33 +15,140 @@ limitations under the License.
+    if (RECOMMENDATION_BAN_VARIANTS.includes(recommendation)) return Recommendation.Ban;
+    if (RECOMMENDATION_OPINION_VARIANTS.includes(recommendation)) return Recommendation.Opinion;
+    return null;
+}
+
+/**
+ * Representation of a rule within a Policy List.
+ */
+export abstract class ListRule {
     /**
-     * The recommendation for this rule, or `null` if there is no recommendation or the recommendation is invalid.
-     * Recommendations are normalised to their stable types.
+     * A glob for `entity`.
      */
-    public get recommendation(): string|null {
-        if (RECOMMENDATION_BAN_TYPES.includes(this.action)) return RECOMMENDATION_BAN;
-        return null;
+    private glob: MatrixGlob;
+    constructor(
+        /**
+         * The entity covered by this rule, e.g. a glob user ID, a room ID, a server domain.
+         */
+        public readonly entity: string,
+        /**
+         * A human-readable reason for this rule, for audit purposes.
+         */
+        public readonly reason: string,
+        /**
+         * The type of entity for this rule, e.g. user, server domain, etc.
+         */
+        public readonly kind: EntityType,
+        /**
+         * The recommendation for this rule, e.g. "ban" or "opinion".
+         */
+        public readonly recommendation: Recommendation | null) {
+        this.glob = new MatrixGlob(entity);
     }
 
+    /**
+     * Determine whether this rule should apply to a given entity.
+     */
     public isMatch(entity: string): boolean {
         return this.glob.test(entity);
     }
 }
+
+/**
+ * A rule representing a "ban".
+ */
+export class ListRuleBan extends ListRule {
+    constructor(
+        /**
+    * The entity covered by this rule, e.g. a glob user ID, a room ID, a server domain.
+    */
+        public readonly entity: string,
+        /**
+         * A human-readable reason for this rule, for audit purposes.
+         */
+        public readonly reason: string,
+        /**
+         * The type of entity for this rule, e.g. user, server domain, etc.
+         */
+        public readonly kind: EntityType,
+    ) {
+        super(entity, reason, kind, Recommendation.Ban)
+    }
+}
+
+/**
+ * A rule representing an "opinion"
+ */
+export class ListRuleOpinion extends ListRule {
+    constructor(
+
+        /**
+ * The entity covered by this rule, e.g. a glob user ID, a room ID, a server domain.
+ */
+        public readonly entity: string,
+        /**
+         * A human-readable reason for this rule, for audit purposes.
+         */
+        public readonly reason: string,
+        /**
+         * The type of entity for this rule, e.g. user, server domain, etc.
+         */
+        public readonly kind: EntityType,
+        /**
+         * A number in [-100, +100] where -100 represents the worst possible opinion
+         * on the entity (e.g. toxic user or community) and +100 represents the best
+         * possible opinion on the entity (e.g. absolute trust).
```

Comment:
woah slow down. This is really important and we cannot get it wrong.

1) Opinion is not the same as trust. An opinion is really a statement about someone's behaviour in the context of room participation (this is different to whether you trust them in the context of moderation), at least that's how i read the MSC. We could wing it later on to propagate the opinions of people we have a good opinion of, and in that case we would be trusting people with positive opinion, but this is still a different use case with its own set of considerations. (That afaik have not been considered yet). So it is much more comforting to leave opinion as distinct from trust and we figure propagation semantics later. 
  
2) If opinion does become synonymous with trust, then you need to figure out the semantics of negative trust in propagation (and also neutral trust?). What does it mean? What happens to a propagation model?

Reply:
Good point, I'll reword.

Reply:
Still uses the words "trusted" and "untrusted" which is wrong, it shouldn't use the word trust at all

Reply:
I thought I had fixed that, my bad.